internal/parser/simple_parser_rules.go

```diff
@@ -3731,11 +3752,17 @@ func (p *simpleParser) parseResultColumn(r reporter) (stmt *ast.ResultColumn) {
 				stmt.Period = period
 			}
 		} else {
+			// Conditions for recursive expressions or single expressions.
 			recExpr := p.parseExprRecursive(&ast.Expr{LiteralValue: tableNameOrAsteriskOrExpr}, r)
 			if recExpr != nil {
 				stmt.Expr = recExpr
 			} else {
-				stmt.Expr = &ast.Expr{LiteralValue: tableNameOrAsteriskOrExpr}
+				singleExpr := p.parseExprBeginWithLiteral(tableNameOrAsteriskOrExpr, r)
+				if singleExpr != nil {
```

Comment:
```suggestion
				if singleExpr := p.parseExprBeginWithLiteral(tableNameOrAsteriskOrExpr, r); singleExpr != nil {
```

Reply:
just a suggestion

Reply:
Sure, but this is an inconsistency in our code base.

Reply:
Some are of this type and some are the traditional ones.